Framework/Algorithms/src/SofQWNormalisedPolygon.cpp

```diff
@@ -311,12 +311,56 @@ void SofQWNormalisedPolygon::init() {
   SofQW::createCommonInputProperties(*this);
 }
 
+/** Checks that the input workspace and table have compatible dimensions
+ * @return a map with the corresponding error messages
+ * describing the problem with the property
+ */
+std::map<std::string, std::string> SofQWNormalisedPolygon::validateInputs() {
+  std::map<std::string, std::string> result;
+  API::MatrixWorkspace_const_sptr inputWS = getProperty("InputWorkspace");
+  if (!inputWS) {
+    result["InputWorkspace"] = "InputWorkspace is of Incorrect type. Please "
+                               "provide a MatrixWorkspace as the "
+                               "InputWorkspace";
+  }
+  TableWorkspace_sptr tableWS = getProperty("DetectorTwoThetaRanges");
+  if (tableWS) {
```

Comment:
This block can be simplified slightly.

The `return` in the first `if` can be removed and the following `if`s converted to `else if`.

It only makes sense to assign `result["DetectorTwoThetaRanges"]` once.